WcaOnRails/app/models/incident.rb

```diff
@@ -38,4 +38,86 @@ def digest_sent_at_consistent
       errors.add(:digest_sent_at, "can't be set if incident is not resolved.")
     end
   end
+
+  def status_class
+    if resolved_at then "success" else "warning" end
+  end
+
+  def status_text
+    if resolved_at then "Resolved" else "Pending" end
+  end
+
+  def digest_class
+    if digest_worthy && !digest_sent_at then "warning" else "success" end
+  end
+
+  def digest_text
+    if digest_worthy
+      if digest_sent_at then "Sent" else "Pending" end
+    else
+      ""
+    end
+  end
+
+  def serializable_hash(options = { :can_view_delegate_matters => false })
+
+    json = {
+      id: id,
+      title: title,
+      url: Rails.application.routes.url_helpers.incident_path(self),
+      status_class: status_class,
+      status_text: status_text,
+      public_summary: public_summary,
+      created_at: created_at,
+      updated_at: updated_at,
+      resolved_at: resolved_at,
+
+      tags: tags,
+      tags_array: tags_array.map { |tag|
+        tags_json = { }
+
+        reg = Regulation.find_or_nil(tag)
+
+        tags_json[:name] = tag
+        tags_json[:label_class] = reg ? "primary" : "default"
+        tags_json[:title] = reg ? link_to(reg["id"], reg["url"], target: "_blank") : tag
+        tags_json[:link] = "<a href='#{Rails.application.routes.url_helpers.incidents_path(tags: tag)}'>Search incidents with this tag</a>"
+
+        if reg
+          tags_json[:link] = `#{reg["content_html"]} <hr/>` + tags_json[:link]
+        end
```

Comment:
I'd prefer to provide the data you're computing here in a more granular way and not squeeze them into one dense `link` field. Your task in `serializable_hash` is to offer data (not HTML!) in a concise way, React can choose if and how it wants to use that data. (See also my other remark about computing tags above)